frontend/src/views/ProfileView.vue

```diff
@@ -35,19 +31,18 @@ export default defineComponent({
 	},
 	data() {
 		return {
+			dataArray: [] as OverviewArray[],
 			user: {} as User,
 		};
 	},
 	async created() {
-		this.user = (await getRequest(`users/${this.profile_id}`)).data;
-		// await useUserStore().refreshAllUsers();
-		// this.user = useUserStore().allUsers.filter((user: User) => (Number(user.id) === Number(this.profile_id)))[0];
-		if (!this.profile_id) {
-			dataArray.value = [
-				{ left: 'this profile', right: 'does not exist' },
-			];
+		await useUserStore().refreshAllUsers();
+		const filteredUsers = useUserStore().allUsers.filter((user: User) => Number(user.id) === Number(this.profile_id));
```

Comment:
you can use `find` here instead of `filter`
find returns 1 instance, not an array, and stops at the first "found" instance. If it is not found the return value is 'undefined'
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find